### ISSUE TITLE
[utils] Add a metric computation feature for detection & recognition

### ISSUE THREAD

**Issue by fg-mindee** [mindee/doctr#109]
For users to confirm the performance we display for pretrained models, they would need an easy way to reproduce these results. As such, I suggest that we add a `doctr/utils/metrics.py` module that will implement this.

As a first implementation, we could try to implement a keras [inherited](https://www.tensorflow.org/api_docs/python/tf/keras/metrics) class for:
- [x] Text detection (#117)
- [x] Text recognition (#110)
- [x] End-to-end OCR (#122)

**Comment by charlesmindee** [comment]
For end-to-end OCR metric, it could be interesting to use a kind of mixed-metric such as in [this repo](https://github.com/faustomorales/keras-ocr/blob/master/keras_ocr/evaluation.py):

- compute IoU between boxes
    - if IoU > detection_thresh, compare sequences
        - if sequences are the same (or lev distance < distance_thresh to smooth it): **true positive**, else **near true positive**
- if some detected boxes are not matching gt boxes or if sequences are not the same, **false positive**
- if some gt boxes are not predicted, **false negative**
With the fps, fns, tps rates we can compute **accuracy, recall and F-score**
